api/src/main/java/net/akami/yggdrasil/api/input/ItemInteractionsListener.java

```diff
@@ -48,23 +49,20 @@ public void onInteract(InteractBlockEvent.Primary event) {
         }
         Optional<ItemStack> optItem = getItem(event);
         optItem.ifPresent((item) ->
-                getHandler(event).ifPresent((handler) -> handler.leftClick(item, event, clock)));
+                getHandler(event).ifPresent((handler) -> handler.leftClick(item, CancellableEvent.of(event), clock)));
     }
 
     @Listener
-    public void onInteract(InteractBlockEvent.Secondary event) {
-        Optional<ItemStack> optItem = getItem(event);
-        if(!optItem.isPresent()) {
-            return;
-        }
-        ItemStack item = optItem.get();
-        if(!item.get(Keys.ITEM_BLOCKSTATE).isPresent()) {
-            return;
-        }
-        getHandler(event).ifPresent((handler) -> handler.rightClick(item, event, clock));
+    public void onChangeBlock(ChangeBlockEvent.Place event) {
+        getItem(event).ifPresent(item -> getHandler(event).ifPresent(handler -> {
+            handler.rightClick(item, CancellableEvent.of(event), clock);
+            if(event.getCause().first(Player.class).map(player -> player.gameMode().get() != GameModes.CREATIVE).orElse(false)) {
+                event.getTransactions().forEach(transaction -> transaction.setValid(false));
+            }
+        }));
     }

```

Comment:
That part is a bit messy with the lambda chain. Could I ask you to divide your code into different methods to use one line lambdas, or even method referencing when possible?